src/rdsn/src/meta/server_state.cpp

```diff
@@ -1267,6 +1267,85 @@ void server_state::drop_app(dsn::message_ex *msg)
     }
 }
 
+void server_state::do_app_rename(std::shared_ptr<app_state> &app, const std::string &old_app_name)
+{
+    auto after_rename_app = [this, app, old_app_name](dsn::error_code ec) mutable {
+        if (ERR_OK == ec) {
+            zauto_write_lock l(_lock);
+            _exist_apps.erase(old_app_name);
+
+            std::vector<dsn::rpc_address> replica_nodes;
+            for (auto &node : _nodes) {
+                if (node.second.alive()) {
+                    replica_nodes.emplace(node.first);
+                }
+            }
+            dsn::error_code err = update_node_configuration(replica_nodes);
+            if (err != ERR_OK) {
+                derror("Failed to update replica_server configuration update immediately, it wiil retry by replica_server selfly later.");
```

Comment:
```suggestion
                derror("Failed to update replica_server configuration update immediately, it will retry by replica_server selfly later.");
```

Reply:
I‘m not sure how does this retry mechanism work, could you please explain about it?